src/shell.c

```diff
@@ -903,6 +902,195 @@ void ShellExecuteCommand(char* p, bool* pbExit)
 	{
 		KePrintMemoryMapInfo();
 	}
+	else if (strcmp (token, "cp") == 0)
+	{
+		LogMsg("Starting to copy...");
+		char* fileNameIn = Tokenize (&state, NULL, " ");
+		if (!fileNameIn)
+		{
+			LogMsg("Usage: cp <input> <output>");
+			return;
+		}
+		if (*fileNameIn == 0)
+		{
+			LogMsg("Usage: cp <input> <output>");
+			return;
+		}
+		
+		char* fileNameOut = Tokenize (&state, NULL, " ");
+		if (!fileNameOut)
+		{
+			LogMsg("Usage: cp <input> <output>");
+			return;
+		}
+		if (*fileNameOut == 0)
+		{
+			LogMsg("Usage: cp <input> <output>");
+			return;
+		}
+		
+		int fd_in = FiOpen (fileNameIn, O_RDONLY);
+		if (fd_in < 0)
+		{
+			LogMsg("cp: Could not open %s for reading: %s", fileNameIn, GetErrNoString(fd_in));
+			return;
+		}
+		
+		int fd_out = FiOpen (fileNameOut, O_WRONLY | O_CREAT | O_TRUNC);
+		if (fd_out < 0)
+		{
+			LogMsg("cp: Could not open %s for writing: %s", fileNameOut, GetErrNoString(fd_out));
+			FiClose(fd_in);
+			return;
+		}
+		
+		LogMsg("Progress...");
+		
+		
+		
+		uint8_t chunk_of_data[MOVEDATA_PRECISION];
+		size_t sz = FiTellSize(fd_in);
+		
+		//write 512 byte blocks
+		for (size_t i = 0; i < sz; i += MOVEDATA_PRECISION)

```

Comment:
This seems fine. However at this point I would create a new function entirely, something like `ShellCopy(pFileNameSrc, pFileNameDst)`.